src/object-store/src/cache_policy.rs

```diff
@@ -31,19 +30,41 @@ use crate::metrics::{
     OBJECT_STORE_LRU_CACHE_MISS,
 };
 
+#[derive(Clone)]
 pub struct LruCacheLayer<C> {
     cache: Arc<C>,
     lru_cache: Arc<Mutex<LruCache<String, ()>>>,
 }
 
-impl<C: Accessor> LruCacheLayer<C> {
-    pub fn new(cache: Arc<C>, capacity: usize) -> Self {
-        Self {
+impl<C: Accessor + Clone> LruCacheLayer<C> {
+    pub async fn new(cache: Arc<C>, capacity: usize) -> Result<Self> {
+        let zelf = Self {
             cache,
             lru_cache: Arc::new(Mutex::new(LruCache::new(
                 NonZeroUsize::new(capacity).unwrap(),
             ))),
+        };
+        zelf.recover_keys().await?;
+
+        Ok(zelf)
+    }
+
+    /// Recover existing keys from `cache` to `lru_cache`.
+    async fn recover_keys(&self) -> Result<()> {
+        let (_, mut pager) = self.cache.list("/", OpList::default()).await?;
+
+        let mut lru_cache = self.lru_cache.lock().await;
+        while let Some(entries) = pager.next().await? {
+            for entry in entries {
+                lru_cache.push(entry.path().to_string(), ());
+            }
         }
+
+        Ok(())
+    }
+
+    pub async fn lru_contains_key(&self, key: &str) -> bool {
```

Comment:
Seems only tests use it
```suggestion
    #[cfg(test)]
    pub async fn lru_contains_key(&self, key: &str) -> bool {
```

Reply:
It's in integration tests, we can't call it if it's tagged by ` #[cfg(test)]`.